lib/orderbook/OrderBook.ts

```diff
@@ -1,151 +1,220 @@
 import uuidv1 from 'uuid/v1';
 
-import OrderBookRepository, { Orders } from './OrderBookRepository';
+import OrderBookRepository from './OrderBookRepository';
 import MatchingEngine from './MatchingEngine';
 import MatchesProcessor from './MatchesProcessor';
 import errors from './errors';
 import Pool from '../p2p/Pool';
 import { orders, matchingEngine, db } from '../types';
 import DB from '../db/DB';
-import { groupBy } from '../utils/utils';
 import Logger from '../Logger';
 import LndClient from '../lndclient/LndClient';
+import { ms } from '../utils/utils';
+import Peer from '../p2p/Peer';
 
 type OrderBookConfig = {
   internalmatching: boolean;
 };
 
+type Orders = {
+  buyOrders: { [ id: string ]: orders.StampedOrder };
+  sellOrders: { [ id: string ]: orders.StampedOrder };
+};
+
 class OrderBook {
+  public pairs: db.PairInstance[] = [];
+  public matchingEngines: { [ pairId: string ]: MatchingEngine } = {};
+
   private logger: Logger = Logger.orderbook;
   private repository: OrderBookRepository;
   private matchesProcessor: MatchesProcessor = new MatchesProcessor();
-  public pairs: db.PairInstance[] = [];
-  public matchingEngines: {[ pairId: string ]: MatchingEngine} = {};
+
+  private ownOrders: { [ pairId: string ]: Orders } = {};
+  private peerOrders: { [ pairId: string ]: Orders } = {};
 
   constructor(private config: OrderBookConfig, db: DB, private pool?: Pool, private lndClient?: LndClient) {
     this.repository = new OrderBookRepository(db);
     if (pool) {
-      pool.on('packet.order', this.addPeerOrder);
+      this.registerListeners(pool);
```

Comment:
i'd rather use the class-level method only for event callback, and have them specified in the constructor, not wrapped inside a different function. (Same as it was with `pool.on('packet.order', this.addPeerOrder)`)